src/pharmpy/tools/structsearch/tool.py

```diff
@@ -88,6 +92,33 @@ def run_tmdd(context, model):
     return res
 
 
+def run_pkpd(context, model):
+    wf = create_fit_workflow(model)
+    task_results = Task('results1', bundle_results)
+    wf.add_task(task_results, predecessors=wf.output_tasks)
+    pk_model = call_workflow(wf, 'results_pd', context)
+
+    pd_direct_models = create_pkpd_models(
+        "direct_effect", model, pk_model[0].modelfit_results.parameter_estimates
+    )
+    pd_comp_models = create_pkpd_models(
+        "effect_compartment", model, pk_model[0].modelfit_results.parameter_estimates
+    )
```

Comment:
A small thing but since these two are always used as `pd_direct_models + pd_comp_models` perhaps it makes more sense that `create_pkpd_models` returns both of them together? Or how come they are split?

Reply:
I was not sure if they will always be used together but you are probably right. That function is only used for the structsearch and thus it makes sense to rewrite the function to return all models together.